src/vcpkg/paragraphs.cpp

```diff
@@ -385,61 +385,79 @@ namespace vcpkg::Paragraphs
 
 namespace
 {
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_any_manifest_text(
+    ExpectedL<SourceControlFileAndLocation> try_load_any_manifest_text(
         StringView text,
         StringView origin,
+        StringView spdx_location,
         MessageSink& warning_sink,
         ExpectedL<std::unique_ptr<SourceControlFile>> (*do_parse)(StringView, const Json::Object&, MessageSink&))
     {
         StatsTimer timer(g_load_ports_stats);
         return Json::parse_object(text, origin).then([&](Json::Object&& object) {
-            return do_parse(origin, std::move(object), warning_sink);
+            return do_parse(origin, std::move(object), warning_sink).map([&](std::unique_ptr<SourceControlFile>&& scf) {
+                return SourceControlFileAndLocation{std::move(scf), origin.to_string(), spdx_location.to_string()};
+            });
         });
     }
 }
 
 namespace vcpkg::Paragraphs
 {
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_project_manifest_text(StringView text,
-                                                                                 StringView origin,
-                                                                                 MessageSink& warning_sink)
+    ExpectedL<SourceControlFileAndLocation> try_load_project_manifest_text(StringView text,
+                                                                           StringView origin,
+                                                                           StringView spdx_location,
+                                                                           MessageSink& warning_sink)
     {
-        return try_load_any_manifest_text(text, origin, warning_sink, SourceControlFile::parse_project_manifest_object);
+        return try_load_any_manifest_text(
+            text, origin, spdx_location, warning_sink, SourceControlFile::parse_project_manifest_object);
     }
 
-    ExpectedL<std::unique_ptr<SourceControlFile>> try_load_port_manifest_text(StringView text,
-                                                                              StringView origin,
-                                                                              MessageSink& warning_sink)
+    ExpectedL<SourceControlFileAndLocation> try_load_port_manifest_text(StringView text,
```

Comment:
Maybe these should not get the 'annotations' since text transforms should not care.